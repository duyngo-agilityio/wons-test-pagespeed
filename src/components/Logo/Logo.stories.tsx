import type { Meta, StoryObj } from '@storybook/react';

// Components
import Logo from './index';

const meta: Meta<typeof Logo> = {
  title: 'Components/Logo',
  component: Logo,
  argTypes: {
    size: {
      description: 'Change the size of the logo',
      options: ['sm', 'lg'],
      control: { type: 'select' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Logo>;

export const Default: Story = {
  args: {
    size: 'lg',
  },
};

export const Small: Story = {
  args: {
    size: 'sm',
  },
};

export const Large: Story = {
  args: {
    size: 'lg',
  },
};
