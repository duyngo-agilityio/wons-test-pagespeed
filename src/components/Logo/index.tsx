// Libs
import { memo } from 'react';
import Link from 'next/link';

// Constants
import { IMAGES, ROUTES } from '@/constants';

// Components
import { ImageFallback } from '@/components';

interface LogoProps {
  size?: 'sm' | 'lg';
}

const Logo = ({ size = 'lg' }: LogoProps): JSX.Element => {
  return (
    <Link href={ROUTES.DASHBOARD}>
      <ImageFallback
        priority
        src={IMAGES.LOGO}
        placeholder="empty"
        alt="wons-logo"
        width={size === 'lg' ? 90 : 42}
        height={size === 'lg' ? 56 : 26}
      />
    </Link>
  );
};

export default memo(Logo);
