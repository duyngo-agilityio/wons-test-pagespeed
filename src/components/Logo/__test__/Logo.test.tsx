// Components
import Logo from '../index';

describe('Logo component', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = testLibJestUtils.render(<Logo />);

    expect(container).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { container } = testLibJestUtils.render(<Logo size="sm" />);

    expect(container).toMatchSnapshot();
  });
});
