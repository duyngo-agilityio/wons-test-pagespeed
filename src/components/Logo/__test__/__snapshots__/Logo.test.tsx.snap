// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component renders correctly 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
     
    <a
      href="/"
    >
      <img
        alt="wons-logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="26"
        src="http://localhost/logo.svg"
        style="color: transparent;"
        width="42"
      />
    </a>
    <div
      class="fixed top-4 right-4"
    />
  </div>
</div>
`;

exports[`Logo component renders correctly and matches snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
     
    <a
      href="/"
    >
      <img
        alt="wons-logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="56"
        src="http://localhost/logo.svg"
        style="color: transparent;"
        width="90"
      />
    </a>
    <div
      class="fixed top-4 right-4"
    />
  </div>
</div>
`;
